components/recent_date.tsx

```diff
@@ -14,37 +15,72 @@ type Props = {
     value: number | Date;
     children?(val: string): React.ReactElement | null;
     intl: IntlShape;
+    useTitleCase?: boolean;
+    dateTimeFormat?: Intl.DateTimeFormatOptions;
 }
 
 class RecentDate extends React.PureComponent<Props> {
     public render() {
-        const {value, timeZone} = this.props;
+        const {
+            value,
+            timeZone,
+            useTitleCase = true,
+            dateTimeFormat,
+            intl
+        } = this.props;
         const date = new Date(value);
 
-        if (isToday(date)) {
-            return (
-                <FormattedMessage
-                    id='date_separator.today'
-                    defaultMessage='Today'
-                />
-            );
-        } else if (isYesterday(date)) {
-            return (
-                <FormattedMessage
-                    id='date_separator.yesterday'
-                    defaultMessage='Yesterday'
-                />
-            );
+        const {is} = fromNow(date);
+
+        if (dateTimeFormat == null) {
+            if (useTitleCase && is.today) {
+                return (
+                    <FormattedMessage
+                        id='date_separator.today'
+                        defaultMessage='Today'
+                    />
+                );
+            } else if (useTitleCase && is.yesterday) {
+                return (
+                    <FormattedMessage
+                        id='date_separator.yesterday'
+                        defaultMessage='Yesterday'
+                    />
+                );
+            } else if (!useTitleCase && is.today) {
+                return (
+                    <FormattedRelativeTime
+                        value={0}
+                        unit='day'
+                        numeric='auto'
+                    />
+                );
+            } else if (!useTitleCase && is.yesterday) {
+                return (
+                    <FormattedRelativeTime
+                        value={-1}
+                        unit='day'
+                        numeric='auto'
+                    />
+                );
+            }
         }
 
-        const options = {
-            timeZone,
-            weekday: 'short',
-            month: 'short',
-            day: '2-digit',
-            year: 'numeric'
-        };
-        const formattedDate = this.props.intl.formatDate(value, options);
+        const options: Intl.DateTimeFormatOptions = dateTimeFormat ?? (
+            is.withinPastSixDays ? {
```

Comment:
Nit: This seems a little long. I am 0/5 though. You call.

Reply:
Can you break this up to use an if statement? I actually thought we didn't allow nested ternary statements